### PR TITLE
[TorchFX] Export to `torch.export.export_for_training`

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Changes

* TorchFX Unit tests are moved from `torch._export.capture_pre_autograd_graph` to `torch.export.export_for_training` 
ALL REFERENCE GRAPHS WERE VALIDATED MANUALLY 
* BC types for `fuse_bn_node` are updated
* NNCFGraphBuilder is updated to support a batch-norm type with only one output node (instead of three)
* Model extractor does not traverse down from constans to prevent redundant nodes in the extracted model when the constant is shared
* `shared_constants_unification_transformation` is removed
* Tests which require `capture_pre_autograd_graph` are removed

### Reason for changes

* To migrate to the lates and recommended export method for TorchFX backend

### Related tickets

#2766 

### Tests

test_shared_constants_unification_not_connected_const
post_training_quantization/540/ is finished successfully

### CODE REVIEW

**Review by alexsu52 — COMMENTED** [review]
Please remove support `capture_pre_autograd` form the tests.

File: tests/torch/fx/test_model_transformer.py [hunk 1]
```diff
@@ -384,17 +380,53 @@ def count_constants(model: torch.fx.GraphModule) -> int:
     return num_constant_nodes
 
 
+@pytest.mark.skipif(sys.platform.startswith("win"), reason="capture_pre_autograd_graph is not supported on Windows")
 def test_create_shared_constant_transformation():
     model = MultiBranchesConnectedModel()
     ex_inputs = torch.ones((1, 3, 3, 3))
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply tests/torch/fx/test_model_transformer.py]: Done

File: tests/torch/conftest.py [hunk 2]
```diff
@@ -120,6 +128,10 @@ def pytest_configure(config: Config):
     if regen_dot:
```
[the remark this answers is not in the document]
Comment (daniil-lyakhov) [reply tests/torch/conftest.py:128]: What if I would like to update json files only?

**Comment by daniil-lyakhov** [comment]
> Please remove support for `capture_pre_autograd` form the tests.

`capture_pre_autograd` is completely removed from unit tests

File: tests/torch/conftest.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anzr299) [reply tests/torch/conftest.py:128]: The `regen_dot` was added there by mistake, I have updated the suggestion.

File: tests/torch/conftest.py [hunk 3]
```diff
@@ -120,6 +128,10 @@ def pytest_configure(config: Config):
...
         os.environ["NNCF_TEST_REGEN_DOT"] = "1"
 
+    regen_json = config.getoption("--regen-json", False)
+    if regen_json:
+        os.environ["NNCF_TEST_REGEN_JSON"] = "1"
+
```
Comment (anzr299) [inline tests/torch/conftest.py:134]:
```suggestion
    for option, var in [("--regen-dot", "NNCF_TEST_REGEN_DOT"), ("--regen-json", "NNCF_TEST_REGEN_JSON")]:
      regen = config.getoption(option, False)
      if regen:
          os.environ[env_var] = "1"

```

Comment (daniil-lyakhov) [reply tests/torch/conftest.py:134]: Done, please check

**Review by anzr299 — APPROVED** [review]

**Pull request merged**